src/js/containers/covid19/Covid19Container.jsx

```diff
@@ -3,76 +3,49 @@
-import { covidPageMetaTags } from 'helpers/metaTagHelper';
+import { useHistory } from 'react-router-dom';
+import GlobalConstants from 'GlobalConstants';
+import { useQueryParams } from 'helpers/queryParams';
 import BaseOverview from 'models/v2/covid19/BaseOverview';
-import {
-    jumpToSection,
-    getStickyBreakPointForCovidBanner,
-    getVerticalOffsetForSidebarFooter
-} from 'helpers/covid19Helper';
-import {
-    slug,
-    getEmailSocialShareData,
-    dataDisclaimerHeight
-} from 'dataMapping/covid19/covid19';
 import { fetchOverview, fetchAwardAmounts } from 'apis/disaster';
 import { useDefCodes } from 'containers/covid19/WithDefCodes';
 import { setOverview, setTotals } from 'redux/actions/covid19/covid19Actions';
-import { showModal } from 'redux/actions/modal/modalActions';
-import DataSourcesAndMethodology from 'components/covid19/DataSourcesAndMethodology';
-import OtherResources from 'components/covid19/OtherResources';
-import Analytics from 'helpers/analytics/Analytics';
-import { componentByCovid19Section } from './helpers/covid19';
-import DownloadButtonContainer from './DownloadButtonContainer';
-import SidebarFooter from '../../components/covid19/SidebarFooter';
+import { defcByPublicLaw } from 'dataMapping/covid19/covid19';
+import Covid19Page from 'components/covid19/Covid19Page';
 
 require('pages/covid19/index.scss');
 
 const Covid19Container = () => {
     const [, areDefCodesLoading, defCodes] = useDefCodes();
-    const [dataDisclaimerBanner, setDataDisclaimerBanner] = useState(Cookies.get('usaspending_data_disclaimer'));
     const overviewRequest = useRef(null);
-    const lastSectionRef = useRef(null);
     const awardAmountRequest = useRef(null);
-    const dataDisclaimerBannerRef = useRef(null);
     const dispatch = useDispatch();
-    const [isBannerSticky, , , setBannerStickyOnScroll] = useDynamicStickyClass(dataDisclaimerBannerRef, getStickyBreakPointForCovidBanner(Cookies.get('usaspending_covid_homepage')));
-
-    const handleScroll = () => {
-        setBannerStickyOnScroll();
-    };
+    const history = useHistory();
+    const { publicLaw } = useQueryParams();
 
     useEffect(() => {
-        window.addEventListener('scroll', handleScroll);
-        return () => {
-            window.removeEventListener('scroll', handleScroll);
-        };
-    }, []);
+        /** Default to all DEFC if:
+         * 1) no public law param is defined
+         * 2) the public law param is invalid
+         * 3) the public law param is for ARP, but the ARP filter is not yet released
+         */
+
+        if (!publicLaw ||
+            !(publicLaw === 'all' || (publicLaw in defcByPublicLaw)) ||
+            (publicLaw === 'american-rescue-plan' && !GlobalConstants.ARP_RELEASED)) {
+            history.replace({
+                pathname: '',
+                search: '?publicLaw=all'
+            });
+        }
+    }, [publicLaw]);
 
     useEffect(() => {
+        let filteredDefc = [];
```

Comment:
instead of using a mutable reference here I recommend we make it a parameter in getOverviewData.